### ISSUE TITLE
proc with no arguments does not pick up current block

### ISSUE THREAD

**Issue by enebo** [jruby/jruby#5596]
This is the base problem of #5595.  Simple one-liner:

```text
jruby -e 'def add_filter(p = proc); p.call(:lol); end; add_filter { |e| p e }'
```

**Comment by enebo** [comment]
I have a simple fix for this...

**Comment by headius** [comment]
This was fixed a different way in 9dd724157dc727df1463afd6f19e10bc7016b9d3.

The `proc` behavior here became a feature in Ruby largely by accident, or in response to the lack of syntax for capturing the current block (`def foo(&block)`). With that syntax added, and given the desire to reduce the number of methods that have "special" visibility into the caller's scope, we do not support calling `proc` without a block of some kind.
